dokan_np/dokannp.c

```diff
@@ -272,10 +272,16 @@ DWORD APIENTRY NPGetConnection(__in LPWSTR LocalName, __out LPWSTR RemoteName,
     return WN_NOT_CONNECTED;
   }
 
+  DWORD currentSessionId;
+  if (!ProcessIdToSessionId(GetCurrentProcessId(), &currentSessionId)) {
+    currentSessionId = 0;
+  }
+
   dosDevice[12] = LocalName[0];
 
   for (unsigned int i = 0; i < nbRead; ++i) {
-    if (wcscmp(dokanMountPointInfo[i].MountPoint, dosDevice) == 0) {
+    if (dokanMountPointInfo[i].SessionId == currentSessionId &&
```

Comment:
I believe `dokanMountPointInfo[i].SessionId` is only set if `DOKAN_EVENT_CURRENT_SESSION` was used
https://github.com/dokan-dev/dokany/blob/master/sys/event.c#L837
Should we here only check if `dokanMountPointInfo[i].SessionId != -1` ?

Reply:
Very good point. I will change that both in `DokanGetMountPointList` and `NPGetConnection`.